pkg/api/errors/placement.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"google.golang.org/grpc/codes"
+
+	kiterrors "github.com/dapr/kit/errors"
+)
+
+func PlacementTableVersionNotEventuallyIncreased(tableVersion int, msg string) error {
```

Comment:
hey @fowlerlee - so you'll want to call this func in place of where the relevant err is thrown. Here is a direct [example of me doing that for the pubsub api](https://github.com/dapr/dapr/pull/7322/files#diff-473161725d4a741e0d28f7ff3f1f8d4eeb201faec147cb1fd31453a7c148c200R131), but youll need to find where this error is thrown in dapr and do the same thing. You'll want to pass in all relevant err details like maybe the appID (which you can typically find off the universal struct) and any other metadata that may be relevant to a user. Specifically, for the kiterrors code - you'll need to open a PR to `dapr/kit` to add the placement `kiterrors.CodePrefixPlacement`[ similar to this PR](https://github.com/dapr/kit/pull/84). Then you'll want to use that kit error prefix for the `.WithErrorInfo()`.

Reply:
@cicoyle : I added the error handling to this PR, and fixed some typos I had that were not conforming to the format tests. The issue still requires integration tests, but I'm not certain the form that these should take, since many already exist? I also 'git rebase' to get the master into the feature branch, hence the many commits from the master to this issue PR.

Reply:
cc @cicoyle